theme/islandora-videojs.tpl.php

```diff
@@ -15,6 +15,9 @@
   <?php foreach ($sources as $source): ?>
     <source src="<?php print $source['url']; ?>" type='<?php print $source['mime']; ?>'>
   <?php endforeach; ?>
+  <?php if ($transcript): ?>
+    <track kind="captions" src="<?php print $transcript_path; ?>" label="<?php print $transcript_language; ?>" default>
```

Comment:
Is... there some form of escaping that should be done on the attribute contents here? Or... maybe back where they're being set in variables, in the `.module`? A `check_plain()` or something? If things somehow contained a `"`... things might go boom? And/or open up the door to adding other arbitrary markup?

Reply:
It seems to be escaped in videojs itself. I tried putting in quotes both closed and not closed, semicolons, and HTML tags both closed and unclosed. Seems to be fine. 

If you have specific concerns about markup that might be put into the VTT file I can test them.

Reply:
Or if you meant the configured DSID? If you attempt to save a DS with invalid characters in the DSID, you get an error. 

So far as I can tell, putting invalid characters, markup, etc. into the variable in the form just results in it not finding the transcript datastream.

Reply:
The label attribute? Especially given it's being scraped from the text? As in, if the "Language" line was something like:
```
Language: labeltext"><script>alert('bad things')</script><span consume="remaining
```

... might have to go further, to terminate the `<track/>` and `<video/>` elements to get it to actually run such a `<script/>` tag, but... is not insurmountable.

Reply:
Aaaah, I see what you mean. I tried adding that language line to the vtt file, and did indeed get the alert.

I added a `check_plain()` to the language matching, and it seems to have worked. Does the implementation make sense?

Reply:
Should do the trick, methinks.